xdem/dem.py

```diff
@@ -84,9 +85,9 @@ def __init__(self, filename_or_dataset, vref_name=None, vref_grid=None, silent=T
         # trying to get vref from product name (priority to user input)
         self.__parse_vref_from_fn(silent=silent)
 
-    def copy(self, new_array=None):
+    def copy(self, new_array: np.ndarray | None = None) -> DEM:
```

Comment:
Simple fix ! But wouldn't it be more generic to use kwargs? This way if one adds new functionalities to the parent function, it is also passed.

Reply:
Ah true, I don't know if that maps the argument for the subclass function tnough.
That's applicable to so many other things, we would have to rethink a lot of Subclasses functions! So not something we want to do in the frame of this PR.

Reply:
Well, you can at least try in that case to replace your argument 'new_array' with **kwargs and see if that solves your issue.
The only problem might be with the docstrings. But one can always refer to the parent's documentation.

Reply:
Just checked, and it works as expected.
Though, to be honest, it seems a lot less practical because the keywords argument are not mapped by the function:
![Screenshot from 2021-11-17 17-13-33](https://user-images.githubusercontent.com/28896516/142238549-60c7c0c2-0be3-4b45-8041-d9777cd95b78.png)
![Screenshot from 2021-11-17 17-13-43](https://user-images.githubusercontent.com/28896516/142238466-2c653fc0-a5fa-48df-8ca5-89dbd1acd480.png)
My opinion is that we should implement this kind of syntax for overriden class function syntax either everywhere or nowhere. Right now it's nowhere in xdem, we repeat arguments everywhere. Copying all arguments is also what the editor recommends when one starts the writing of such an overriden parent class function.
So it is best to raise an issue and tackle this consistently and all at once in another PR.